File: tshirt-store-backend/Models/Order.cs
namespace MerchStoreBackend.Models;

public class Order
{
    public int Id { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string ShippingAddress { get; set; } = string.Empty;
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public string PaymentMethod { get; set; } = string.Empty;
    public string? CardNumber { get; set; }
    public string? CardHolder { get; set; }
    public string? ExpiryDate { get; set; }
    public string? CVV { get; set; }
    public List<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: tshirt-web-store/src/api/order.ts
import axios from 'axios';

export interface OrderItem {
  productId: number;
  quantity: number;
  customText?: string;
  textColor?: string;
  textPosition?: string;
}

export interface PaymentDetails {
  method: 'COD' | 'Card';
  cardNumber?: string;
  cardHolder?: string;
  expiryDate?: string;
  cvv?: string;
}

export interface Order {
  customerName: string;
  shippingAddress: string;
  payment: PaymentDetails;
  items: OrderItem[];
}

const API_BASE = 'http://localhost:5286/api/order';

export async function createOrder(order: Order): Promise<void> {
  await axios.post(API_BASE, order);
  console.log('Order created successfully');
}
